packages/shared/lib/services/proxy.service.unit.test.ts

```diff
@@ -420,6 +420,46 @@ describe('Proxy service Construct URL Tests', () => {
         expect(url).toBe('https://myinstanceurl.com/api/test');
     });
 
+    it('Should handle Proxy base URL interpolation where connection configuration param is present', () => {
+        const config = {
+            template: {
+                auth_mode: AuthModes.OAuth2,
+                proxy: {
+                    base_url: '${connectionConfig.api_base_url_for_customer} || https://api.gong.io'
```

Comment:
I don't think I have the full context but I would not go this way and add a new attributes to avoid this pseudo-js syntax. I'll let the others weigh on this